comixed-web/src/app/library/effects/comic.effects.spec.ts

```diff
@@ -0,0 +1,124 @@
+ * Copyright (C) 2020, The ComiXed Project
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses>
+ */
+
+import { Observable, of, throwError } from 'rxjs';
+import { ComicEffects } from '@app/library/effects/comic.effects';
+import { ComicDetailsService } from '@app/library/services/comic-details.service';
+import { AlertService } from '@app/core';
+import { TestBed } from '@angular/core/testing';
+import { TranslateModule } from '@ngx-translate/core';
+import { LoggerModule } from '@angular-ru/logger';
+import { provideMockActions } from '@ngrx/effects/testing';
+import { COMIC_1 } from '@app/library/library.fixtures';
+import {
+  comicSaved,
+  saveComicDetails,
+  saveComicDetailsFailed
+} from '@app/library/actions/comic-details.actions';
+import { hot } from 'jasmine-marbles';
+import { HttpErrorResponse } from '@angular/common/http';
+import {
+  HttpClientTestingModule,
+  HttpTestingController
+} from '@angular/common/http/testing';
+import { MatSnackBarModule } from '@angular/material/snack-bar';
+import { Comic } from '@app/library';
+
+describe('ComicEffects', () => {
+  let actions$: Observable<any>;
+  let effects: ComicEffects;
+  let comicDetailsService: jasmine.SpyObj<ComicDetailsService>;
+  let alertService: AlertService;
+  let httpMock: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        TranslateModule.forRoot(),
+        LoggerModule.forRoot(),
+        HttpClientTestingModule,
+        MatSnackBarModule
+      ],
+      providers: [
+        ComicEffects,
+        provideMockActions(() => actions$),
+        {
+          provide: ComicDetailsService,
+          useValue: {
+            saveComic: jasmine.createSpy('ComicDetailsService.saveComic()')
+          }
+        },
+        AlertService
+      ]
+    });
+    httpMock = TestBed.inject(HttpTestingController);
+    effects = TestBed.inject(ComicEffects);
+    comicDetailsService = TestBed.inject(ComicDetailsService) as jasmine.SpyObj<
+      ComicDetailsService
+    >;
+    alertService = TestBed.inject(AlertService);
+    spyOn(alertService, 'info');
+    spyOn(alertService, 'error');
+  });
+
+  it('should be created', () => {
+    expect(effects).toBeTruthy();
+  });
+
+  describe('when saving comic detail', () => {
+    it('fire an action on success', () => {
+      const serviceResponse = COMIC_1 as Comic;
```

Comment:
The "as Comic" piece here isn't necessary, since COMIC_1 is already an instance of the Comic interface.

That said, can we instead declare a constant named COMIC at the top of this testing suite and use that in all the tests instead? That way we can re-use it for other tests that'll be written in future.